### PR TITLE
[Pipeline](shared_scan_opt) Support shared scan opt in pipeline exec engine

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

clickbench 1 tablet：

image

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.8 seconds
 stream load tsv:          478 seconds loaded 74807831229 Bytes, about 149 MB/s
 stream load json:         41 seconds loaded 2358488459 Bytes, about 54 MB/s
 stream load orc:          69 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          28 seconds loaded 861443392 Bytes, about 29 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230227161544_clickbench_pr_105205.html

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run allbuild

**Comment by HappenLee** [comment]
run allbuild

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

**Comment by HappenLee** [comment]
run buildall

File: be/src/vec/exec/scan/pip_scanner_context.h [hunk 1]
```diff
@@ -33,20 +33,68 @@ class PipScannerContext : public vectorized::ScannerContext {
...
+            _blocks_queue_empty.emplace_back(true);
+            _queue_mutexs.emplace_back(new std::mutex);
+            _blocks_queues.emplace_back(std::list<vectorized::BlockUPtr>());
+        }
+    }
 
 private:
-    std::atomic_bool _blocks_queue_empty = true;
+    int _next_queue_to_feed = 0;
+    std::vector<bool> _blocks_queue_empty;
```
Comment (wangbo) [inline be/src/vec/exec/scan/pip_scanner_context.h:95]: What is ```_blocks_queue_empty``` used for ?
It seems that no place using it?

Comment (HappenLee) [reply be/src/vec/exec/scan/pip_scanner_context.h:95]: will remove next pr

File: be/src/common/status.h [hunk 2]
```diff
@@ -402,6 +404,8 @@ class Status {
 
     bool ok() const { return _code == ErrorCode::OK; }
 
+    bool is_blocked_by_sc() const { return _code == ErrorCode::PIP_WAIT_FOR_SC; }
```
Comment (Gabriel39) [inline be/src/common/status.h:407]: use is<ErrorCode::PIP_WAIT_FOR_SC>() instead of is_blocked_by_sc

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**